client_operations.go

```diff
@@ -200,11 +436,7 @@ func (list *ListAccountsParams) URLParams() []KeyValue {
 func (c *Client) ListAccounts(params *ListAccountsParams) *AccountList {
 	path := "/accounts"
 	path = BuildUrl(path, params)
-	return &AccountList{
-		client:       c,
-		nextPagePath: path,
-		HasMore:      true,
-	}
+	return NewAccountList(c, path)
```

Comment:
Decided to make constructors for these to make mocking easier

Reply:
The programmer can pass in an HttpCaller here instead of a client.

Reply:
yeah i think this is definitely the right call and it keeps our encapsulation nice as well